### PR TITLE
Refine mirror settings

### PR DESCRIPTION
**Pull request by blindpirate**
- Only set plugin portal mirror on CI
- Use mirror for EC2 agents and macOS agents

### CODE REVIEW

File: gradle/shared-with-buildSrc/mirrors.settings.gradle.kts [hunk 1]
```diff
@@ -71,7 +69,7 @@ fun normalizeUrl(url: String): String {
 
 fun overridesPluginPortalUrl() = providers.systemProperty(PLUGIN_PORTAL_OVERRIDE_URL_PROPERTY).forUseAtConfigurationTime().orNull != null
 
-if (!overridesPluginPortalUrl() && !isEc2Agent() && !isMacAgent() && !ignoreMirrors()) {
+if (isCI() && !overridesPluginPortalUrl() && !ignoreMirrors()) {
     // https://github.com/gradle/gradle-private/issues/2725
     // https://github.com/gradle/gradle-private/issues/2951
     System.setProperty(PLUGIN_PORTAL_OVERRIDE_URL_PROPERTY, "https://repo.grdev.net/artifactory/gradle-plugin-portal-prod/")
```
Comment (guylabs) [inline gradle/shared-with-buildSrc/mirrors.settings.gradle.kts:75]: Can't we pass the URL via an environment variable?

**Review by guylabs — CHANGES_REQUESTED** [review]

**Comment by guylabs** [comment]
@blindpirate keep in mind that all changes you did/plan to do should also be done on the `release` branch, as repo.gradle.org will be updated once we migrated and then all branches need to be adapted to the new Artifactory instance.

File: gradle/shared-with-buildSrc/mirrors.settings.gradle.kts (hunk 1 above)
Comment (blindpirate) [reply gradle/shared-with-buildSrc/mirrors.settings.gradle.kts]: I just noticed we have already defined `%gradle.plugins.portal.url%` in Root project, so we can just remove it.

**Review by guylabs — APPROVED** [review]

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Review by wolfs — COMMENTED** [review]
LGTM!

**Pull request merged**